test/parallel_api/numeric/numeric.ops/in_place_scan.pass.cpp

```diff
@@ -19,64 +19,331 @@
 
 #include "support/test_config.h"
 #include "support/utils.h"
+#include "support/scan_serial_impl.h"
 
 #if TEST_DPCPP_BACKEND_PRESENT
-#    include "support/utils_sycl.h"
-#    include "support/sycl_alloc_utils.h"
-#    include "support/scan_serial_impl.h"
+#include <CL/sycl.hpp>
+#endif
+using namespace TestUtils;
+
+#if TEST_DPCPP_BACKEND_PRESENT
+static const char kMsgInclusiveScanNormal [] = "Wrong effect from inclusive scan (non-inplace)";
+static const char kMsgInclusiveScanInplace[] = "Wrong effect from inclusive scan (inplace)";
+static const char kMsgExclusiveScanNormal [] = "Wrong effect from exclusive scan (non-inplace)";
+static const char kMsgExclusiveScanInplace[] = "Wrong effect from exclusive scan (inplace)";
+
+// TODO: replace data generation with random data and update check to compare result to
+// the result of a serial implementation of the algorithm
+template <typename Iterator1, typename Size>
+void
+initialize_data(Iterator1 host_keys, Size n)
+{
+    const Size kStartVal = 1;
+    for (Size i = 0; i != n; ++i)
+        host_keys[i] = kStartVal + i;
+}
+
+struct TestingAlgoritmInclusiveScan
+{
+    template <typename... TArgs>
+    void
+    call_onedpl(TArgs&&... args)
+    {
+        oneapi::dpl::inclusive_scan(::std::forward<TArgs>(args)...);
+    }
+
+    template <typename... TArgs>
+    void
+    call_serial(TArgs&&... args)
+    {
+        inclusive_scan_serial(::std::forward<TArgs>(args)...);
+    }
+
+    const char*
+    getMsg(bool bInplace) const
+    {
+        return bInplace ? kMsgInclusiveScanInplace : kMsgInclusiveScanNormal;
+    }
+};
+
+template <typename BinaryOp, int InitValue>
+struct TestingAlgoritmInclusiveScanExt
+{
+    template <typename... TArgs>
+    void
+    call_onedpl(TArgs&&... args)
+    {
+        oneapi::dpl::inclusive_scan(::std::forward<TArgs>(args)..., BinaryOp(), InitValue);
+    }
+
+    template <typename... TArgs>
+    void
+    call_serial(TArgs&&... args)
+    {
+        inclusive_scan_serial(::std::forward<TArgs>(args)..., BinaryOp(), InitValue);
+    }
+
+    const char*
+    getMsg(bool bInplace) const
+    {
+        return bInplace ? kMsgInclusiveScanInplace : kMsgInclusiveScanNormal;
+    }
+};
+
+template <int InitValue>
```

Comment:
To tell the truth it not good design to pass `init `value via template petameter...
In that case we cannot various that value in runtime, in a loop for example (if we will want to extend the test coverage).

Reply:
We are able to pass additional parameters into test only through template arguments because instance of test class created deep inside of ```test2buffers``` and ```test1buffer``` function calls.